cylc/rose/platform_utils.py

```diff
@@ -47,15 +49,31 @@ def get_platform_from_task_def(
     config = WorkflowConfig(flow, flow_file, Values())
     # Get entire task spec to allow Cylc 7 platform from host guessing.
     task_spec = config.pcfg.get(['runtime', task])
+    # check for subshell and evaluate
+    if (
+        task_spec.get('platform', None) is not None and
+        is_platform_definition_subshell(task_spec['platform'])
```

Comment:
In some cases parsec might give us an empty string rather than None. For example you get an empty string if you try to unset the platform like so:

```
[runtime]
    [[task]]
        platform =
```

So I think it should be safer to do `if platform_string` than `if platform_string is None`:

```suggestion
        task_spec.get('platform')
        and is_platform_definition_subshell(task_spec['platform'])
```